src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -1968,6 +1974,143 @@ public Task inspectTask(final String taskId) throws DockerException, Interrupted
   }
 
   @Override
+  public List<Config> listConfigs() throws DockerException, InterruptedException {
+    assertApiVersionIsAbove("1.30");
+
+    WebTarget resource = resource().path("configs");
+
+    try {
+      return request(GET, CONFIG_LIST, resource, resource.request(APPLICATION_JSON_TYPE));
+    } catch (DockerRequestException e) {
+      switch (e.status()) {
+        case 503:
+          throw new NonSwarmNodeException("node is not part of a swarm", e);
+        default:
+          throw e;
+      }
+    }
+  }
+
+  @Override
+  public List<Config> listConfigs(Config.Criteria criteria)
+      throws DockerException, InterruptedException {
+    assertApiVersionIsAbove("1.30");
+
+    final Map<String, List<String>> filters = new HashMap<>();
+
+    if (criteria.configId() != null) {
+      filters.put("id", Collections.singletonList(criteria.configId()));
+    }
+    if (criteria.label() != null) {
+      filters.put("label", Collections.singletonList(criteria.label()));
+    }
+    if (criteria.name() != null) {
+      filters.put("name", Collections.singletonList(criteria.name()));
```

Comment:
@davidxia I've tried using the 'names' filter but docker 17.06 throws an error whenever I try and specify it so I only implemented the 'name' option because that appears to work. I'll try and dig into it some more to figure out what's going on.